plugins/inputs/mesos/mesos.go

```diff
@@ -425,22 +455,22 @@ type TaskStats struct {
 	Statistics  map[string]interface{} `json:"statistics"`
 }
 
-func (m *Mesos) gatherSlaveTaskMetrics(address string, defaultPort string, acc telegraf.Accumulator) error {
+func (m *Mesos) gatherSlaveTaskMetrics(address, username, password string, acc telegraf.Accumulator) error {
 	var metrics []TaskStats
-
-	host, _, err := net.SplitHostPort(address)
-	if err != nil {
-		host = address
-		address = address + defaultPort
-	}
+	address, host := prepareAddress(address)
 
 	tags := map[string]string{
 		"server": host,
 	}
 
-	ts := strconv.Itoa(m.Timeout) + "ms"
+	requestURL := fmt.Sprintf("%s/monitor/statistics?timeout=%sms", address, strconv.Itoa(m.Timeout))
```

Comment:
if you return *url.URL from prepareAddress, you can construct this full request URI using Go builtin URL parsing. See [url.Values](https://golang.org/pkg/net/url/#Values) and [url.URL.Path](https://golang.org/pkg/net/url/#URL)